hikari/impl/event_factory.py

```diff
@@ -490,12 +501,23 @@ def deserialize_audit_log_entry_create_event(
     # INTERACTION EVENTS #
     ######################
 
+    _interaction_events_map: dict[
+        base_interactions.InteractionType, typing.Type[interaction_events.InteractionCreateEvent]
+    ] = {
+        base_interactions.InteractionType.APPLICATION_COMMAND: interaction_events.CommandInteractionCreateEvent,
+        base_interactions.InteractionType.AUTOCOMPLETE: interaction_events.AutocompleteInteractionCreateEvent,
+        base_interactions.InteractionType.MESSAGE_COMPONENT: interaction_events.ComponentInteractionCreateEvent,
+        base_interactions.InteractionType.MODAL_SUBMIT: interaction_events.ModalInteractionCreateEvent,
+    }
+
     def deserialize_interaction_create_event(
         self, shard: gateway_shard.GatewayShard, payload: data_binding.JSONObject
     ) -> interaction_events.InteractionCreateEvent:
-        return interaction_events.InteractionCreateEvent(
-            shard=shard, interaction=self._app.entity_factory.deserialize_interaction(payload)
-        )
+        interaction = self._app.entity_factory.deserialize_interaction(payload)
+
+        if event := self._interaction_events_map.get(interaction.type):
+            return event(shard=shard, interaction=interaction)
+        raise errors.UnrecognisedEntityError(f"Unrecognised interaction type {interaction.type}")
```

Comment:
```suggestion
        event = _interaction_events_map[interaction.type]
        return self._interaction_events_map[interaction.type](shard=shard, interaction=interaction)
```

If we ever get here and we do not deserialize it properly, its because we didnt update `_interaction_events_map` (and, as such, a keyerror), not because the entity was not recognised